compiler/rustc_middle/src/lint.rs

```diff
@@ -132,31 +132,38 @@ impl ShallowLintLevelMap {
         let mut owner = start.owner;
         let mut specs = &self.specs;
 
-        for parent in tcx.hir().parent_id_iter(start) {
-            if parent.owner != owner {
-                owner = parent.owner;
-                specs = &tcx.shallow_lint_levels_on(owner).specs;
-            }
-            if let Some(map) = specs.get(&parent.local_id)
-                && let Some(&(level, src)) = map.get(&id)
-            {
-                return (Some(level), src);
+        if let Some(tcx) = tcx {
+            for parent in tcx.hir().parent_id_iter(start) {
+                if parent.owner != owner {
+                    owner = parent.owner;
+                    specs = &tcx.shallow_lint_levels_on(owner).specs;
+                }
+                if let Some(map) = specs.get(&parent.local_id)
+                    && let Some(&(level, src)) = map.get(&id)
+                {
+                    return (Some(level), src);
+                }
             }
         }
 
         (None, LintLevelSource::Default)
     }
 
     /// Fetch and return the user-visible lint level for the given lint at the given HirId.
-    #[instrument(level = "trace", skip(self, tcx), ret)]
+    #[instrument(level = "trace", skip(self, tcx, sess), ret)]
     pub fn lint_level_id_at_node(
         &self,
-        tcx: TyCtxt<'_>,
+        tcx: Option<TyCtxt<'_>>,
```

Comment:
Is this change a leftover from an old revision?

Reply:
no. we need to call this function during linking, when a tcx isn't available. see https://github.com/rust-lang/rust/pull/119286/files#diff-862130c0af71054886913befc8084094485fa7d764d95f55644e033e34acd126R1061-R1063

Reply:
The `rustc_middle::lint::lint_level` function already doesn't accept a `TyCtxt` anyway. Every caller of `lint_level_id_at_node` calls it with `Some(tcx)`.

Reply:
yes, you're right, sorry. i've reverted these changes.